registry/machine.go

```diff
@@ -91,9 +102,61 @@ func (r *EtcdRegistry) SetMachineState(ms machine.MachineState, ttl time.Duratio
 		return uint64(0), err
 	}
 
+	// Set the initial metadata on creation
+	if err = r.setMachineMetadata(ms.ID, ms.Metadata); err != nil {
+		return uint64(0), err
+	}
+
 	return resp.Node.ModifiedIndex, nil
 }
 
+func (r *EtcdRegistry) SetMachineMetadata(machID string, key string, value string) error {
+	//Attempt to update key
```

Comment:
One last nit-picky thing: comments are `// <Comment text>`. Note the trailing space after the forward slashes and the first character of the comment body is uppercase. It's arbitrary, but let's at least be consistent.

This comment specifically isn't actually necessary, since you are no longer attempting to update the key, you are setting the key.